jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/FormFields.java

```diff
@@ -75,11 +75,26 @@ public static CompletableFuture<Fields> from(Request request, Charset charset)
         return from(request, charset, maxFields, maxLength);
     }
 
+    public static void set(Request request, CompletableFuture<Fields> fields)
+    {
+        request.setAttribute(FormFields.class.getName(), fields);
+    }
+
+    public static CompletableFuture<Fields> get(Request request)
+    {
+        Object attr = request.getAttribute(FormFields.class.getName());
+        if (attr instanceof FormFields futureFormFields)
+            return futureFormFields;
+        return EMPTY;
+    }
+
```

Comment:
The reason they exist is that they hide the magic of how the form fields are associated with a request.  It is currently done via an attribute, but ultimately we might provide a field for it, or some other mechanism.
They deal with `CompletableFuture` because that's how FormFields are obtained, in that they may be read asynchronously.  It just so happens that servlets use a blocking get.